app/2.3.x/external-plugins.md

```diff
@@ -0,0 +1,247 @@
+
+property | description | default
+---------|-------------|--------
+`pluginserver_<NAME>_socket` | unix socket path | `/usr/local/kong/<NAME>.socket`
+`pluginserver_<NAME>_start_cmd` | command to start the plugin server process | `/usr/local/bin/<NAME>`
+`pluginserver_<NAME>_query_cmd` | command to dump available plugins' info | `/usr/local/bin/query_<NAME>`
+
+
+For example, you could set Go and Python plugins like this:
+
+```
+pluginserver_names = go,python
+
+pluginserver_go_socket = /usr/local/kong/go_pluginserver.socket
+pluginserver_go_start_cmd = go-pluginserver -kong-prefix /usr/local/kong/ -plugins-dir /usr/local/kong/go-plugins
+pluginserver_go_query_cmd = go-pluginserver -dump-all-plugins -plugins-dir /usr/local/kong/go-plugins
+
+pluginserver_python_socket = /usr/local/kong/pyton_pluginserver.socket
+pluginserver_python_start_cmd = pypluginserver.py
+pluginserver_python_query_cmd = pypluginserver.py -dump
+```
+
+### Legacy configuration
+
+Kong versions 2.0 to 2.2 supported only Go external plugins and a single
+pluginserver using a different configuration style.  Starting with Kong 2.3,
+the old style is recognized and internally transformed to the new style.
+
+If property `pluginserver_names` isn't defined, the legacy properties
+`go_plugins_dir` and `go_pluginserver_exe` are tried:
+
+property | description | default
+---------|-------------|--------
+`go_plugins_dir` | directory with Go plugins | `off`, meaning to disable Go plugins
+`go_pluginserver_exe` | path to the go-pluginserver executable | `/usr/local/bin/go-pluginserver`
+
+Notes:
+
+- The old style doesn't allow multiple plugin servers.
+- Version 0.5.0 of [go-pluginserver] requires the old style configuration.
+- The new style configuration requires v0.6.0 of [go-pluginserver]
+
+## Developing Go plugins
+
+Kong support for the Go language consist of two parts:
+
+- [go-pdk] as a library, provides Go functions to access Kong features of the [PDK][kong-pdk].
+- [go-pluginserver] an executable to dynamically load plugins written in Go.
+
+
+Notes:
+
+The Kong version 2.3 allows multiple plugin servers; in particular
+it's now possible to write "single-plugin servers", in effect plugins as
+microservices.  To help on this, version v0.6.0 of the [go-pdk] package
+includes an optional plugin server, see [Embedded Server](#embedded-server)
+for info on this.
+
+The [go-pluginserver] process is still supported.  Its main advantage is
+that it is a single process for any numbe of plugins, but the dynamic
+loading of plugins has proven challenging under Go language; unlike
+the microservice architecture, which is well supported by the language
+and tools.
+
+### Development
+
+To write a Kong plugin in Go, you need to:
+
+1. Define a structure type to hold configuration.
+2. Write a `New()` function to create instances of your structure.
+3. Add methods on that structure to handle phases.
+   
+if you want a dynamically-loaded plugin to be used with [go-pluginserver]:
```

Comment:
```suggestion
If you want a dynamically-loaded plugin to be used with [go-pluginserver]:
```

Reply:
You'll probably need to indent to avoid breaking the numbering sequence. @javierguerragiraldez Can you please provide a direct review link to this content so we can check?

Reply:
the first three steps are the same, but from there it diverges.  the "if" lines should break the flow but keep the numbers.   didn't find a more clear way to put it without drawing a flow diagram